app/controllers/ems_common.rb

```diff
@@ -477,6 +477,9 @@ def button
                           :action     => "edit",
                           :id         => find_checked_ids_with_rbac(CloudVolume).first
     elsif params[:pressed] == "cloud_volume_delete"
+      # Clear CloudVolumeController's lastaction, since we are calling the delete_volumes from
+      # an external controller. This will ensure that the final redirect is properly handled.
+      session["#{CloudVolumeController.session_key_prefix}_lastaction".to_sym] = nil
```

Comment:
Interesting approach :)
Maybe it's just me, but the syntax seems a bit cryptic.
Although, I must say, it's the most logical approach to make `@lastaction` = `nil` in this scenario.

I was thinking along the lines of a little shuffling around the `if`/`elsif` conditional in `delete_volumes`
```diff
--- a/app/controllers/cloud_volume_controller.rb
+++ b/app/controllers/cloud_volume_controller.rb
@@ -458,11 +458,11 @@ class CloudVolumeController < ApplicationController
       @single_delete = true unless flash_errors?
       if @flash_array.nil?
         add_flash(_("The selected %{model} was deleted") % {:model => ui_lookup(:table => "cloud_volume")})
+      else
+        drop_breadcrumb(:name => 'dummy', :url => " ") # missing a bc to get correctly back so here's a dummy
+        session[:flash_msgs] = @flash_array.dup if @flash_array
+        redirect_to(previous_breadcrumb_url)
       end
-    else
-      drop_breadcrumb(:name => 'dummy', :url => " ") # missing a bc to get correctly back so here's a dummy
-      session[:flash_msgs] = @flash_array.dup if @flash_array
-      redirect_to(previous_breadcrumb_url)
     end
   end
```

LMK what you think of the above approach and we will take it from there.